src/main/mediaPermissions.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0
+ * Vesktop, a desktop app aiming to give you a snappier Discord Experience
+ * Copyright (c) 2023 Vendicated and Vencord contributors
+ */
+
+import { session, systemPreferences } from "electron";
+
+export function registerMediaPermissionsHandler() {
+    if (process.platform !== "darwin") return;
+
+    session.defaultSession.setPermissionRequestHandler((_webContents, permission, callback, details) => {
+        if (permission === "media") {
+            if (details.mediaTypes!.includes("audio")) {
```

Comment:
no reason not to just use safer optional chaining
```suggestion
            if (details.mediaTypes?.includes("audio")) {
```